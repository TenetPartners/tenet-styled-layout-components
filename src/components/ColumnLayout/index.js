import styled from 'styled-components';

export default styled.div`
  display: flex;
  > * {
    flex-basis: ${props => 100 / props.children.length}%;
  }
`;
