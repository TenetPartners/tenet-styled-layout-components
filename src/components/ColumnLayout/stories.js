/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable jsx-a11y/media-has-caption */
import React from 'react';
import { storiesOf } from '@storybook/react';
import ColumnLayout from '.';

storiesOf('ColumnLayout', module)
  .add('with 2 columns', () => (
    <ColumnLayout>
      <p>hello</p>
      <p>world</p>
    </ColumnLayout>
  ))
  .add('with 3 columns', () => (
    <ColumnLayout>
      <p>hello</p>
      <p>world</p>
      <p>3rd column</p>
    </ColumnLayout>
  ))
  .add('with 2 videos', () => (
    <ColumnLayout>
      <video controls="" poster="//d166gs6ce8o9t3.cloudfront.net/Tenet_Corp.jpg">
        <source src="//d166gs6ce8o9t3.cloudfront.net/Tenet_Corp.mp4" type="video/mp4" />
        <source src="//d166gs6ce8o9t3.cloudfront.net/Tenet_Corp.webm" type="video/webm" />
      </video>
      <video controls="" poster="//d166gs6ce8o9t3.cloudfront.net/Tenet_Corp.jpg">
        <source src="//d166gs6ce8o9t3.cloudfront.net/Tenet_Corp.mp4" type="video/mp4" />
        <source src="//d166gs6ce8o9t3.cloudfront.net/Tenet_Corp.webm" type="video/webm" />
      </video>
    </ColumnLayout>
  ));
