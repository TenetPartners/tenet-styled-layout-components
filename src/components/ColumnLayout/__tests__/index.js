/* eslint-disable function-paren-newline */
import React from 'react';
import renderer from 'react-test-renderer';
import ColumnLayout from '../index';

describe('ColumnLayout', () => {
  it('renders correctly for 2 columns', () => {
    const tree = renderer.create(
      <ColumnLayout>
        <p>hello</p>
        <p>world</p>
      </ColumnLayout>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('flex-basis', '50%', {
      modifier: '> *',
    });
  });

  it('renders correctly for 4 columns', () => {
    const tree = renderer.create(
      <ColumnLayout>
        <p>hello</p>
        <p>world</p>
        <p>1</p>
        <p>2</p>
      </ColumnLayout>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('flex-basis', '25%', {
      modifier: '> *',
    });
  });
});
