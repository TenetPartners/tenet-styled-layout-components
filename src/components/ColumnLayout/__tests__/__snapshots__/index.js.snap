// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnLayout renders correctly for 2 columns 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > * {
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
}

<div
  className="c0"
>
  <p>
    hello
  </p>
  <p>
    world
  </p>
</div>
`;

exports[`ColumnLayout renders correctly for 4 columns 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > * {
  -webkit-flex-basis: 25%;
  -ms-flex-preferred-size: 25%;
  flex-basis: 25%;
}

<div
  className="c0"
>
  <p>
    hello
  </p>
  <p>
    world
  </p>
  <p>
    1
  </p>
  <p>
    2
  </p>
</div>
`;
